### PR TITLE
Mentoring and coaching

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/MentoringAndCoaching/controller.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Request, Response } from "express";
+import { searchMentorShipService } from "./services";
+
+export const search = async (req: Request, res: Response) => {
+  const { data, status = 200 } = await searchMentorShipService(req.body);
+
+  return res.status(status).send({ data });
```
Comment (emindsajay) [inline src/MentoringAndCoaching/controller.ts]: Please remove the data key in send call. We are sending data directly.

File: src/MentoringAndCoaching/schema-build-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+export const buildContext = async (input: any = {}): Promise<any> => {
+  const context: MentoringContext = {
+    domain: `${process.env.DOMAIN}${input?.category ?? "mentoring"}`,
+    action: input.action ?? "",
+    bap_id: process.env.BAP_ID || (input?.bap_id ?? ""),
+    bap_uri: process.env.BAP_URI || (input?.bap_uri ?? ""),
+    timestamp: input.timestamp ?? moment().toISOString(),
+    message_id: input?.messageId ?? uuid(),
+    version: process.env.CORE_VERSION || (input?.core_version ?? ""),
+    ttl: "PT10M", // ask Ajay for its significance
```
Comment (emindsajay) [inline src/MentoringAndCoaching/schema-build-helpers.ts:14]: I think we need this for protocal server.

File: src/MentoringAndCoaching/schema-build-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import moment from "moment";
+import { v4 as uuid } from "uuid";
+import { MentoringContext } from "./schema";
+
+export const buildContext = async (input: any = {}): Promise<any> => {
```
Comment (emindsajay) [inline src/MentoringAndCoaching/schema-build-helpers.ts]: Was there any reason to make this function **async** ? If not any speicific, can we remove this ?

File: src/MentoringAndCoaching/controller.ts (hunk 1 above)
Comment (ghost) [reply src/MentoringAndCoaching/controller.ts]: done

File: src/MentoringAndCoaching/schema-build-helpers.ts (hunk 3 above)
Comment (ghost) [reply src/MentoringAndCoaching/schema-build-helpers.ts]: done

**Pull request merged**